### DISCUSSION TITLE
Compressing FermiNet by 330x using JAX (FermiKAN PoC)

### DISCUSSION THREAD

**Discussion by Samu-Physicist** [jax-ml/jax#41020] [Show and tell]
<img width="40%" alt="compression_chart" src="https://github.com/user-attachments/assets/64a4fce9-e073-4103-9f55-95eaf0b38871" />

Hi everyone! I’m an R&D researcher at a chemical manufacturer. I usually rely on commercial quantum chemistry solvers, but Neural QMC caught my eye—theoretically, it offers highly accurate quantum mechanics evaluation at $\mathcal{O}(N^4)$ computational cost!

I started experimenting with DeepMind’s FermiNet and successfully reached chemical accuracy for H2 and LiH systems. That got me thinking: what if we replaced the MLPs with KANs (Kolmogorov-Arnold Networks) acting as a learnable physical basis set on single-electron orbitals? This could potentially demonstrate a "Glass-Box" AI, highly interpretable and compatible with LLMs (Neuro-Symbolic approach).

Thanks to JAX's incredible expressiveness, implementing these complex basis function expansions was elegant. By replacing branching B-splines with orthogonal polynomials and utilizing jnp.einsum, I was able to compile the KAN architecture into GEMMs. I built this entirely as a weekend project, pairing with Gemini as my coding partner to push the limits of what I could do with JAX.

Here are the results from my H2 PoC:

 - The Good: Achieved a 330x parameter compression compared to the original FermiNet. We also gained fascinating interpretability regarding the physical basis sets.
 - The Bad (The Wall): I fell into a dependency hell with the K-FAC optimizer and my KAN implementation, so I had to fall back to Adam.
 - The Trade-offs: The accuracy is currently about ~0.01 Hartree shy of the original FermiNet. In an apples-to-apples comparison (using Adam), it took 4x the iterations to converge.
 
<img width="75%" alt="learning_curve" src="https://github.com/user-attachments/assets/88271b32-8988-41e6-8e46-6c45a9b10aac" />

This is a personal, non-peer-reviewed project. I’ve written a detailed breakdown of the architecture (including 4D manifold embeddings), the math, and my struggles here:

👉 [Link to GitHub Repo](https://github.com/Samu-Physicist/FermiKAN)

👉 [Link to Hashnode Article](https://samu-physicist.hashnode.dev/tpu-friendly-kolmogorov-arnold-networks-for-neural-qmc-a-330x-compression-of-ferminet-using-jax)

I’d love to hear your thoughts! I’m especially hoping to collaborate with JAX/XLA wizards (I’d really appreciate some advice on overcoming the K-FAC integration issue) and domain experts!
